test/CloudBoilerplateNet.Tests/Services/CachedDeliveryClientTests.cs

```diff
@@ -192,6 +192,20 @@ public CachedDeliveryClientTests()
             Assert.Equal(1, actualHttpRequests.Value);
         }
 
+        [Fact]
+        public async void AvoidsAddingNullIdentifierTokens()
+        {
+            var cachedClient = GetCachedDeliveryClient(() =>
+            {
+                mockHttp.When($"{baseUrl}/items/coffee_beverages_explained")
+                .Respond("application/json", File.ReadAllText(Path.Combine(Environment.CurrentDirectory, "Fixtures\\CachedDeliveryClient\\item.json")));
+            });
+
+            var item = (await cachedClient.GetItemAsync<Article>("coffee_beverages_explained")).Item;
+
+            Assert.NotNull(item);
```

Comment:
I would add assertion that the dependencies does not contains any null identifiers   

something like:
var identifiers = cachedClient.GetMyIdentifiers (there are method for loading them).
Assert.Empty(identifiers.select(identifier => identifier == null))

What do you think?

Reply:
The identifiers, as well as the dependencies, are kept as method locals. It would be cumbersome to rewrite the CachedDeliveryClient (or the cache manager) to just allow to test for the presence of them. The culprit of #58 was the lack of `parameters`, therefore checking of the dependencies (as you propose) would not help IMO.

If `parameters` were `null` and the `identifierTokens.AddNonNullRange` failed, the current test method would end up with an `ArgumentNullException`. I think the method serves its purpose, given the current shape of the client class.

Or, did you mean something else?